src/containers/Application/ApplicationPage.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react'
+import { FullStructure, ResponsesByCode } from '../../utils/types'
+import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
+import { ApplicationStatus } from '../../utils/generated/graphql'
+import { useApplicationState } from '../../contexts/ApplicationState'
+import { useUserState } from '../../contexts/UserState'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { Loading, NoMatch } from '../../components'
+import strings from '../../utils/constants'
+import messages from '../../utils/messages'
+import { Button, Grid, Header, Segment, Sticky } from 'semantic-ui-react'
+
+interface ApplicationProps {
+  structure: FullStructure
+  responses?: ResponsesByCode
+}
+
+const ApplicationPage: React.FC<ApplicationProps> = ({ structure }) => {
+  const [isStrictPage, setIsStrictPage] = useState(null)
+  const { error, isLoading, fullStructure, responsesByCode } = useGetFullApplicationStructure(
+    structure
+  )
+  const {
+    userState: { currentUser },
+  } = useUserState()
+  const { push } = useRouter()
+
+  console.log('Structure', fullStructure)
+
+  useEffect(() => {
+    if (!structure) return
+
+    // Re-direct based on application status and progress
+    if (structure.info.current?.status === ApplicationStatus.ChangesRequired)
+      push(`/application/${structure.info.serial}`)
+    if (structure.info.current?.status !== ApplicationStatus.Draft)
+      push(`/application/${structure.info.serial}/summary`)
+
+    // TO-DO: Redirect based on Progress (wait till Progress calculation is done)
+  }, [structure])
+
+  if (error) return <NoMatch />
```

Comment:
I think Nicole had a different output if structure is undefined -> (404) and a way to display errors set in the hook. @nmadruga  ?

Reply:
Yup, got it.
```
if (error) return <Message error header={strings.ERROR_APPLICATION_PAGE} list={[error]} />
```

2bebe02